### ISSUE TITLE
Trying to replicate the spack-stack environment on Derecho for testing

### ISSUE THREAD

**Issue by uturuncoglu** [JCSDA/spack-stack#1093]
**Describe the bug**
I am trying to install spack-stack 1.5.1 on Derecho but it does not concretize and not create spack.lock under my environment and have message like in the install step

```
==> /glade/work/turuncu/COP/spack-stack/envs/unified-env environment has no specs to install
```

I also tried with the develop and it is same. Maybe I am missing something in here but not sure. I am plaining to install the tools with newer version of ESMF library to test some ideas and development. Maybe it is platform specific issue and I wonder if you have any idea?

**To Reproduce**
```
module purge
module use /glade/work/epicufsrt/contrib/spack-stack/derecho/modulefiles
module load ecflow/5.8.4
module load mysql/8.0.33
module use /lustre/desc1/scratch/epicufsrt/contrib/modulefiles_extra
export LMOD_TMOD_FIND_FIRST=yes
export SPACK_ROOT=/glade/work/turuncu/COP/spack-stack/spack
source setup.sh
spack stack create env --site derecho --template ufs-weather-model --name unified-env --prefix /glade/work/turuncu/COP/spack-stack/envs/unified-env/install
cd envs/unified-env
spack env activate .
cd ../..
spack --color always concretize
spack --color always install --source -j3
```

**Expected behavior**
Need to install all the tools to new environment.

**System:**
NCAR's Derecho with Intel compiler

**Additional context**
No

**Comment by climbfuji** [comment]
1. You name your env unified-dev but use template ufs-weather-model, is that intentional? This shouldn't be the cause of your problem, but normally we use the env name unified-env for template unified-dev.
2. I think you are missing the step where you edit envs/my-env-name/spack.yaml and comment out compilers you don't want. See https://github.com/JCSDA/spack-stack/blob/develop/configs/templates/unified-dev/spack.yaml - probably that's not needed for ufs-weather-model template though.

**Comment by uturuncoglu** [comment]
@climbfuji Thanks. Yes, probably I would be nice to change the name of the environment to prevent confusion. Let me check your suggestion about the compiler. There was no any warning or error from the spack about what was wrong. So, it is hard to predict that there are multiple compilers and not working because of it. Anyway, thanks for your help. I'll try again and update you.

**Comment by RatkoVasic-NOAA** [comment]
@uturuncoglu please compare these 9 files with your files:

/glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.5.1/envs/unified-env/
spack.yaml

/glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.5.1/envs/unified-env/common/
config.yaml
modules.yaml
packages.yaml

/glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.5.1/envs/unified-env/site/
compilers.yaml
config.yaml
mirrors.yaml
modules.yaml
packages.yaml

**Comment by uturuncoglu** [comment]
@RatkoVasic-NOAA In my case spack yaml is totally different. I have following

```
# spack-stack hash: eac2ad4
# spack hash: b106aa8c34
spack:
  concretizer:
    unify: when_possible

  config:
    install_tree:
      root: /glade/work/turuncu/COP/spack-stack/envs/ufs-weather-model/install
  modules:
    default:
      roots:
        lmod: /glade/work/turuncu/COP/spack-stack/envs/ufs-weather-model/install/modulefiles
        tcl: /glade/work/turuncu/COP/spack-stack/envs/ufs-weather-model/install/modulefiles
  view: false
  include:
  - site
  - common

  specs:
  - ufs-weather-model-env~debug
```

I also tried @climbfuji suggestion and remove unrelevent sections related with the compiler and it is same. Maybe it is related with the `ufs-weather-model` template. I tried with unified-dev template and now spack.yaml is very similar (I am not sure but hates are different, see below). 

```
turuncu@derecho1/glade/work/turuncu/COP/spack-stack/envs/unified-env (release/1.5.1) $ diff /glade/work/epicufsrt/contrib/spack-stack/derecho/spack-stack-1.5.1/envs/unified-env/spack.yaml spack.yaml 
1,2c1,2
< # spack-stack hash: a677421
< # spack hash: 12690f08aa
---
> # spack-stack hash: eac2ad4
> # spack hash: b106aa8c34
13c13
<   - compilers: ['%gcc', '%intel']
---
>   - compilers: ['%aocc', '%apple-clang', '%gcc', '%intel']
35a36
>     - fms@2023.03
59,61c60,67
<   - fms@2023.03%intel
<   - fms@2023.03%gcc
<   mirrors: {}
---
>   config:
>     install_tree:
>       root: /glade/work/turuncu/COP/spack-stack/envs/unified-env/install
>   modules:
>     default:
>       roots:
>         lmod: /glade/work/turuncu/COP/spack-stack/envs/unified-env/install/modulefiles
>         tcl: /glade/work/turuncu/COP/spack-stack/envs/unified-env/install/modulefiles
```

The issue that I have could be specific to `ufs-weather-model` template. I just wonder if you are testing that regularly on every platforms. I think installing `unified-dev` will take longer since it has also installation related with Jedi etc. So, it would be nice to have capability to install `ufs-weather-model` one.

**Comment by RatkoVasic-NOAA** [comment]
@uturuncoglu no, we are not testing all environments. That would take a long time. We test unified one, that covers almost all (except GSI additions where needed).
I know it will take longer, but that should be done only once (hopefully).

**Comment by climbfuji** [comment]
@uturuncoglu Any update on this? Were you able to install your environment?

**Comment by RatkoVasic-NOAA** [comment]
@uturuncoglu I did it on Hera two ways
1) using unified-dev template and reducing number of apps being installed:
/scratch2/NCEPDEV/fv3-cam/Ratko.Vasic/SS/spack-stack.1.5.1/envs/unified-dev
2) using ufs-weather-model-static environment:
/scratch2/NCEPDEV/fv3-cam/Ratko.Vasic/SS/spack-stack.1.5.1/envs/ufs-weather-model-static
You can see my changes to
     common/config.yaml
     common/modules.yaml
     common/packages.yaml
     site/compilers.yaml
     site/config.yaml
     site/mirrors.yaml
     site/modules.yaml
     site/packages.yaml
     spack.yaml
files.
With both I ran WM and regression tests passed.
NOTE, for speedup, I used Intel only.

**Comment by uturuncoglu** [comment]
@climbfuji I could not find time to look at it and I go with standalone esmf installation manually.

**Comment by uturuncoglu** [comment]
@RatkoVasic-NOAA Thanks for your help. Unfortunately, I don't have access to Hera. I tried to use same template by modifying its content but end up to same situation on Derecho. Then, decided to install ESMF manually. Thanks again for your help. If you want you could close this ticket since I have manual esmf installation at this point.
